backend/fallback_manager.py

```
@@ -233,6 +233,34 @@ def get_mode_string(self) -> str:
+        try:
+            target_mode = FallbackMode(mode)
+            if target_mode != self.current_mode:
+                # Create synthetic health state for transition logging
+                synthetic_state = {
+                    "circuit_breaker": {},
+                    "retry": {},
+                    "system": {},
+                }
+                await self._transition_to_mode(target_mode, synthetic_state)
+            logger.debug(f"Fallback mode set to {mode}")
+            return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)